homeassistant/components/downloader/services.py

```diff
@@ -8,15 +8,20 @@
 import threading
 
 import requests
+from requests.auth import HTTPBasicAuth, HTTPDigestAuth
 import voluptuous as vol
 
+from homeassistant.const import HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION
```

Comment:
```suggestion
from homeassistant.const import CONF_AUTHENTICATION, CONF_PASSWORD, CONF_USERNAME, HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION
```

Reply:
> HA conventions are to use CONF_*** for configuration (config entry data or YAML configuration files), and reserve ATTR_*** for actions or state attributes.

Are there docs that discuss this convention? The suggestion would seem to conflict with this convention:

My concern with using `CONF_AUTHENTICATION` versus `ATTR_AUTH_TYPE` would be that _auth type_, _username_, and _password_ are all _action_ parameters, not integration _configuration_ parameters.

While those constants exist on _homeassistant.const_, the incidence of _CONF\__ for accessing `service.data` fields is quite low...maybe 3 integrations return hits for the regex `service\.data\[CONF_`. The majority use _CONF\__ to access `entry.data`. Similarly, the incidence of `entry\.data\[ATTR_` is almost zero.